api/v6/report_server.thrift

```diff
@@ -250,6 +251,13 @@ struct LinesInFilesRequested {
 }
 typedef list<LinesInFilesRequested> LinesInFilesRequestedList
 
+struct SourceComponentData {
+  1: string name,        // Name of the source component.
+  2: string value,       // Value of the source component.
```

Comment:
I think these comments does not add any information to the source code. A better comment would describe how the possible values might look like (are the filters comma separated? do we need to escape commas? Or if we have multiple filters, why don't we store the values as a list of strings? Also, is value a good name for a source component? I think something like a filter_list would describe this better.).

Reply:
In the database we store the value of the source component as one string.
"+/a/b
-/a/c"
I think the name `value` is a good name in this case.

We can handle these lines separately and create a separate database to store value differently.
`Components`:
- `name`: unique name of the source component.
- `component_value_id`: Reference to ComponentValues table.

`ComponentValues`
- `id`
- `type` - enum value skip (`-`) or include (`+`)
- `file_path` - path glob pattern

In this case we can handle values of the source component separately and return a list of object which stores the type (skip or include) and file path glob pattern.
Eg.
```thrift
enum SourceComponentFilterType {
 SKIP,
 INCLUDE
}

struct SourceComponentFilterValue {
  1: SourceComponentFilterType type,
  2: string filePath
}

struct SourceComponentData {
  1: string name,
  2: list<SourceComponentFilterValue> values
}
```